src/HTMLGenerator/converter.py

```diff
@@ -14,40 +15,47 @@ def _get_path_relative_to_username(filename):
     filename_path = Path(filename)
     return str(filename_path.relative_to(s.expanduser()))
 
-def convert(top_level_output_directory, top_level_input_directory, json_with_filepaths, path_to_current_working_directory = Path.cwd()):    
-    converter_path = Path(path_to_current_working_directory / "ConvertFromConfigurationFile.vi")
+def write_configuration_file(json_with_filepaths, top_level_output_directory = Path(""), top_level_input_directory = Path(""), path_to_current_working_directory = Path.cwd()):
     current_configuration_absolute_path = Path(path_to_current_working_directory / CURRENT_CONFIGURATION_PATH)
-    notification_absolute_path = Path(path_to_current_working_directory / NOTIFICATION_PATH)
-    call_converter_command = str(LABVIEW_PATH) + " " + str(converter_path)
-    try:
-        if os.path.isfile(notification_absolute_path):
-            os.remove(notification_absolute_path)                
-    except:
-        pass
     # Writing the top, preload, input and output path to the currentConfiguration.txt file.
     try:
         with open(current_configuration_absolute_path, 'w') as current_configuration_file:
-            current_configuration_file.write(
+            if "topPath" in json_with_filepaths:
+                current_configuration_file.write(
                 '/top:' + str(top_level_output_directory / json_with_filepaths["topPath"]) + "\n")
-            current_configuration_file.write(
+            if "preloadedFiles" in json_with_filepaths:
+                current_configuration_file.write(
                 '/preload:' + str(Path(json_with_filepaths["preloadedFiles"])) + "\n")
-            current_configuration_file.write(
+            if "inputPath" in json_with_filepaths:
+                current_configuration_file.write(
                 '/input:' + str(top_level_input_directory / json_with_filepaths["inputPath"]) + "\n")
```

Comment:
Isn't this the same thing all over again? The only difference is that now the `top_level_output_directory` and `top_level_input_directory` have some default values. This still doesn't separate out the way we create the config file for the test-suite vs the generic converter, right?

The main reason for separating out the two ways of writing currentConfiguration.txt was so that the (generic) converter doesn't need to know about the `top_level_output_directory` and `top_level_input_directory` values. These are very specific to the test suite, the converter does not need to know about it at all. All the converter.py script should care about, is creating a currentConfiguration.txt _only and only_ from whatever JSON file it gets. It should not have to worry about what the `top_level_output_directory` and `top_level_input_directory` are.
The test suite on the other hand is trying to make use of the converter with it's own modifications, which require the the `top_level_output_directory` and `top_level_input_directory`.

Since the converter itself doesn't have to know about the `top_level_output_directory` and `top_level_input_directory`, should it even care about those?

I'm not quite sure about this. @rajneesh What is your opinion about this? Should the sequencer and converter have their own separate versions of `write_configuration_file()`? Or should they reuse the same code inside the converter (which would mean the converter now has to know about two extra parameters, even though they have empty default values)?

Reply:
Maybe the JSON structure itself should be changed to more appropriately reflect the values that are accepted by GeneratorUI-Advanced.vi. That way, we can just reuse the same JSON structure while converting multiple projects as well. We can change it to this format, as used in [this](https://rajneeshlakkundi.visualstudio.com/FINALE/_git/FINALE.typescript/commit/7f70e5f3e2d8bc83fe80232fe4be0e43c26e35f9?refName=refs%2Fheads%2Finlakkur%2Fcli-support&path=%2Fsrc%2Fcli%2FinputFiles.json) branch, like this:

{
    "top": "C:\\FINALE\\2018-09-04\\src",
    "configurations": [
        {
            "preload": [ "Path of the project to preload", "One more project to preload" ],
            "input": "Path to source file/directory that needs to be converter",
            "output": "Destination path relative to the top-level output"
        },
        {
            "preload": [ ],
            "input": "Path to source file/directory that needs to be converter",
            "output": "Destination path relative to the top-level output"
        },
        ...
    ]
 }

That way, this can also work for converting multiple projects at the same time.